homeassistant/components/life360/device_tracker.py

```diff
@@ -0,0 +1,617 @@
+CONF_ERROR_THRESHOLD = 'error_threshold'
+CONF_HOME_PLACE = 'home_place'
+CONF_MAX_GPS_ACCURACY = 'max_gps_accuracy'
+CONF_MAX_UPDATE_WAIT = 'max_update_wait'
+CONF_MEMBERS = 'members'
+CONF_PLACES = 'places'
+CONF_SHOW_AS_STATE = 'show_as_state'
+CONF_WARNING_THRESHOLD = 'warning_threshold'
+
+AZ_ONCE = 'once'
+SHOW_DRIVING = 'driving'
+SHOW_MOVING = 'moving'
+SHOW_PLACES = 'places'
+SHOW_AS_STATE_OPTS = [SHOW_DRIVING, SHOW_MOVING, SHOW_PLACES]
+
+ATTR_ADDRESS = 'address'
+ATTR_AT_LOC_SINCE = 'at_loc_since'
+ATTR_DRIVING = SHOW_DRIVING
+ATTR_LAST_SEEN = 'last_seen'
+ATTR_MOVING = SHOW_MOVING
+ATTR_RADIUS = 'radius'
+ATTR_RAW_SPEED = 'raw_speed'
+ATTR_SPEED = 'speed'
+ATTR_WAIT = 'wait'
+ATTR_WIFI_ON = 'wifi_on'
+
+SERVICE_ZONES_FROM_PLACES = 'zones_from_places'
+
+EVENT_UPDATE_OVERDUE = 'life360_update_overdue'
+EVENT_UPDATE_RESTORED = 'life360_update_restored'
+
+
+def _excl_incl_list_to_filter_dict(value):
+    return {
+        'include': CONF_INCLUDE in value,
+        'list': value.get(CONF_EXCLUDE) or value.get(CONF_INCLUDE)
+    }
+
+
+def _prefix(value):
+    if not value:
+        return ''
+    if not value.endswith('_'):
+        return value + '_'
+    return value
+
+
+def _thresholds(config):
+    error_threshold = config.get(CONF_ERROR_THRESHOLD)
+    warning_threshold = config.get(CONF_WARNING_THRESHOLD)
+    if error_threshold and warning_threshold:
+        if error_threshold <= warning_threshold:
+            raise vol.Invalid('{} must be larger than {}'.format(
+                CONF_ERROR_THRESHOLD, CONF_WARNING_THRESHOLD))
+    elif not error_threshold and warning_threshold:
+        config[CONF_ERROR_THRESHOLD] = warning_threshold + 1
+    elif error_threshold and not warning_threshold:
+        # Make them the same which effectively prevents warnings.
+        config[CONF_WARNING_THRESHOLD] = error_threshold
+    else:
+        # Log all errors as errors.
+        config[CONF_ERROR_THRESHOLD] = 1
+        config[CONF_WARNING_THRESHOLD] = 1
+    return config
+
+
+_SLUG_LIST = vol.All(
+    cv.ensure_list, [cv.slugify],
+    vol.Length(min=1, msg='List cannot be empty'))
+
+_LOWER_STRING_LIST = vol.All(
+    cv.ensure_list, [vol.All(cv.string, vol.Lower)],
+    vol.Length(min=1, msg='List cannot be empty'))
+
+_EXCL_INCL_SLUG_LIST = vol.All(
+    vol.Schema({
+        vol.Exclusive(CONF_EXCLUDE, 'incl_excl'): _SLUG_LIST,
+        vol.Exclusive(CONF_INCLUDE, 'incl_excl'): _SLUG_LIST,
+    }),
+    cv.has_at_least_one_key(CONF_EXCLUDE, CONF_INCLUDE),
+    _excl_incl_list_to_filter_dict,
+)
+
+_EXCL_INCL_LOWER_STRING_LIST = vol.All(
+    vol.Schema({
+        vol.Exclusive(CONF_EXCLUDE, 'incl_excl'): _LOWER_STRING_LIST,
+        vol.Exclusive(CONF_INCLUDE, 'incl_excl'): _LOWER_STRING_LIST,
+    }),
+    cv.has_at_least_one_key(CONF_EXCLUDE, CONF_INCLUDE),
+    _excl_incl_list_to_filter_dict
+)
+
+_THRESHOLD = vol.All(vol.Coerce(int), vol.Range(min=1))
+
+LIFE360_SCHEMA = PLATFORM_SCHEMA.extend({
```

Comment:
Instead of configuring it as a platform, let's configure it under the DOMAIN, so it will be easier to transition to a config entry in the future.

Reply:
How about the work has already been done, let's just use it as-is and stop trying to put more work on me? It works like most other device trackers. It fits into the existing system just fine. If you want to do it some other way, then how about adding an issue to change it as part of another task? E.g., how about doing that as part of changing device_tracker to a platform component?